### PR TITLE
Support env variables in yml

### PR DESCRIPTION
**Pull request by ardatan**
#1302

### CODE REVIEW

File: packages/graphql-codegen-cli/tests/cli-flags.spec.ts [hunk 1]
```diff
@@ -151,4 +151,17 @@ describe.only('CLI Flags', () => {
     const config = createConfig(args);
     expect(config.overwrite).toBeTruthy();
   });
+
+  it('Should interpolate environmental variables in YML', () => {
+    process.env['SCHEMA_PATH'] = 'schema-env.graphql';
+    mockConfig(`
+        schema: \${SCHEMA_PATH}
```
Comment (dotansimha) [inline packages/graphql-codegen-cli/tests/cli-flags.spec.ts]: @ardatan why not just `$SOME_VAR`?

Comment (ardatan) [reply packages/graphql-codegen-cli/tests/cli-flags.spec.ts]: In the issue, the interpolated string has `{}.
`${API_ENDPOINT}`

Comment (dotansimha) [reply packages/graphql-codegen-cli/tests/cli-flags.spec.ts]: I know, but I think `$API_ENDPOINT` is better API (similar to usage in terminal), and shouldn't conflict with other syntax.

File: packages/graphql-codegen-cli/src/yml.ts [hunk 2]
```diff
@@ -1,7 +1,19 @@
 import { Types, debugLog } from 'graphql-codegen-core';
 import { safeLoad } from 'js-yaml';
 
+export function interpolate<T extends { [key: string]: string }>(str: string, context: T) {
+  // tslint:disable-next-line:forin
+  for (let key in context) {
```
Comment (kamilkisiela) [inline packages/graphql-codegen-cli/src/yml.ts]: Maybe we should replace all variables outside of `context` with `undefined`?

Something like: `/($([a-z0-9\-_]))/g` and replace `$ASD` with `process.env.ASD` which will give an undefined if value is not available.

Comment (kamilkisiela) [reply packages/graphql-codegen-cli/src/yml.ts]: Now we iterate through all properties of `process.env` and if there's a variable that is not a part of that we end up with a string, `$ASD` for example.

File: packages/graphql-codegen-cli/tests/cli-flags.spec.ts (hunk 1 above)
Comment (ardatan) [reply packages/graphql-codegen-cli/tests/cli-flags.spec.ts]: But I think the standard is the other one.

File: packages/graphql-codegen-cli/src/yml.ts (hunk 2 above)
Comment (ardatan) [reply packages/graphql-codegen-cli/src/yml.ts]: Done :)

Comment (kamilkisiela) [reply packages/graphql-codegen-cli/src/yml.ts]: I don't trust `.*` but if it works (as test shows) then I'm okay with it :))

**Comment by jordmccord** [comment]
The following codegen.yml does not work, just returns an empty string I've tried with .env and also with the command `graphql-codegen -r dotenv/config`. Any ideas why?
```
schema:
  - ${SCHEMA_URL}:
      headers:
        Authorization: ${BEARER_TOKEN}
  - "src/client/client-schema.graphql"
documents:
  - "src/mutations/*.ts"
  - "src/queries/*.ts"
  - "src/subscriptions/*.ts"
generates:
  ./src/types/apolloTypes.tsx:
    plugins:
      - typescript
      - typescript-operations
      - typescript-react-apollo
    config:
      withMutationFn: false
      withComponent: false
      withHOC: false
      withHooks: true
      scalars:
        DateTime: string
  ./src/client/fragmentTypes.json:
    plugins:
      - fragment-matcher

```

**Comment by ardatan** [comment]
@jordmccord See here https://github.com/dotansimha/graphql-code-generator/issues/2722#issuecomment-542172410

**Pull request merged**